### PR TITLE
refa(design): label component redesign

### LINKED ISSUE #717: Design & component update - Label
Current Version:
- simple Label component with htmlFor prop
- used in Checkbox, Radio, Field and Textarea component

-> [Figma Design](https://www.figma.com/file/DFKyTGHAoDxOsUBPszLLxP/%F0%9F%8F%B5%EF%B8%8FMarigold?node-id=467%3A104)

**Todo**:
- remove the required prop from Checkbox, Radio, Field and Textarea component. Add it to the Label component.
- place the label optionally above or inline with the referring components
- add `variant: 'section'`
- add `variant: 'collective'`

### PR DESCRIPTION
**Pull request by ti10le**
closes #717 

changings in Radio, Checkbox, Field and Textarea because of the placement from the Label und the new required prop in Label component.

### CODE REVIEW

File: packages/components/src/Label/Label.stories.mdx [hunk 1]
```diff
@@ -23,12 +23,28 @@ import { Label } from './Label';
...
     },
+    required: {
+      control: {
+        type: 'boolean',
+      },
+      options: [true, false],
+      description: 'Adds required Icon to label',
+      table: {
+        defaultValue: {
+          summary: 'false',
```
Comment (viktoria-schwarz) [inline packages/components/src/Label/Label.stories.mdx:34]: 'false' without the ticks :)

**Review by viktoria-schwarz — APPROVED** [review]

File: docs/content/components/label.mdx [hunk 2]
```diff
@@ -7,14 +7,15 @@ title: Label
...
 ## Properties
 
-| Property             | Type     | Default   |
-| :------------------- | :------- | :-------- |
-| `variant` (optional) | `string` | `default` |
-| `htmlFor`            | `string` |           |
+| Property              | Type      | Default |
+| :-------------------- | :-------- | :------ |
+| `variant` (optional)  | `string`  | `above` |
+| `required` (optional) | `boolean` |         |
```
Comment (viktoria-schwarz) [inline docs/content/components/label.mdx:17]: defaults to False

File: packages/components/src/Label/Label.stories.mdx [hunk 3]
```diff
@@ -23,12 +23,28 @@ import { Label } from './Label';
         required: true,
       },
     },
+    required: {
+      control: {
+        type: 'boolean',
+      },
+      options: [true, false],
```
Comment (viktoria-schwarz) [inline packages/components/src/Label/Label.stories.mdx:30]: no need to set options if Type is boolean

File: themes/theme-b2b/src/index.ts [hunk 4]
```diff
@@ -377,6 +384,13 @@ const theme: BaseTheme = {
       lineHeight: 'body',
       color: colors.gray30,
     },
+    section: {
+      fontFamily: 'body',
+      fontSize: 'small',
+      fontWeight: 'body',
+      lineHeight: '32px',
```
Comment (viktoria-schwarz) [inline themes/theme-b2b/src/index.ts:391]: why don't you use lineHeight from theme?

Comment (ti10le) [reply themes/theme-b2b/src/index.ts:391]: line height from theme is just:
`lineHeights: {
    body: 1.5,
    heading: 1.125,
  },`

32px like in the figma design is 2rem

Comment (viktoria-schwarz) [reply themes/theme-b2b/src/index.ts:398]: We actually use 32px lineHeight a lot in the theme, maybe we should think about adding 1.5rem and 2rem to the theme, but not in this issue :)

Comment (ti10le) [reply themes/theme-b2b/src/index.ts:398]: yes good idea!

**Review by viktoria-schwarz — APPROVED** [review]

**Pull request merged**